### PR TITLE
Add e2e test for storing and listing rules

### PR DESCRIPTION
**Pull request by onprem**
Signed-off-by: Prem Saraswat <prmsrswt@gmail.com>

### CODE REVIEW

File: test/e2e/configs.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+) {
+	yamlContent := []byte(fmt.Sprintf(
+		objstoreYamlTpl,
+		bucket,
+		accessKey,
+		secretKey,
+		endpoint,
+	))
+
+	err := ioutil.WriteFile(
```
Comment (bwplotka) [inline test/e2e/configs.go:55]:
```suggestion
	testutil.Ok(t, ioutil.WriteFile(
```

File: test/e2e/readwrite_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	tenantA := "tenant_a"
+	tenantB := "tenant_b"
+
+	t.Run("valid-rules-read-write", func(t *testing.T) {
+		rctx, cancel := context.WithTimeout(context.Background(), time.Second*20)
+		t.Cleanup(cancel)
+
+		// Retrying the first request as minio takes some time to get ready, even after readiness check passes.
+		// Details: https://github.com/efficientgo/e2e/issues/11.
+		err = runutil.Retry(time.Second*3, rctx.Done(), func() error {
```
Comment (bwplotka) [inline test/e2e/readwrite_test.go:92]:
```suggestion
		testutil.Ok(t, runutil.Retry(time.Second*3, rctx.Done(), func() error {
```

File: test/e2e/readwrite_test.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+			if res.StatusCode/100 != 2 {
+				return fmt.Errorf("statuscode expected 200, got %d", res.StatusCode)
+			}
+
+			return nil
+		})
+		testutil.Ok(t, err)
+
+		res, err := client.SetRulesWithBody(ctx, tenantB, "application/yaml", strings.NewReader(sampleRulesB))
```
Comment (bwplotka) [inline test/e2e/readwrite_test.go:106]: Just curious, how tenant param is used right now by server? (:

**Review by bwplotka — APPROVED** [review]
Looks good, just tiny style nits, great job! I  see there is... some tenancy (:

File: test/e2e/readwrite_test.go (hunk 3 above)
Comment (onprem) [reply test/e2e/readwrite_test.go:106]: So the server gets this tenant param from request URL, and then it creates/updates the object under `/<tenant>/rules.yaml` in object storage. 

The files in object storage looks something like:
```
- tenant_a
- - rules.yaml
- tenant_b
- - rules.yaml
```

Comment (bwplotka) [reply test/e2e/readwrite_test.go:105]: Thanks!

**Review by bwplotka — APPROVED** [review]

File: test/e2e/readwrite_test.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+var invalidRules = `
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - record: trs
+        expr: vector(1)
+      - invalid: property`
+
+func TestMetricsReadAndWrite(t *testing.T) {
```
Comment (matej-g) [inline test/e2e/readwrite_test.go:59]: Might be a little bit too much copying going on :stuck_out_tongue_winking_eye: Maybe:
```suggestion
func TestRulesReadAndWrite(t *testing.T) {
```

**Review by matej-g — APPROVED** [review]

**Pull request merged**